[KB-177] Problem definition based on mathematical definitions of the problem area (#124)
chhoumann/thesis-chemcam: report_thesis/src/sections/problem_definition.tex

\section{Problem Definition}\label{sec:problem_definition}
The primary objective of this research is to enhance computational methods for the accurate and robust quantification of chemical compositions using \gls{libs} data.
This enhancement seeks to improve the adaptability and precision of \gls{libs} across various environmental contexts, including terrestrial and planetary geological samples.

> 

Reviewer: We've mentioned multicollinearity and matrix effects in the introduction. This sentence reads a little like its the first time we've presented it. Does it perhaps make sense to rephrase it so that its clear that its not the first time we mention it?

> Quantifying chemical compositions from \gls{libs} spectral data poses significant challenges due to the high dimensionality of the data and the complex physical and chemical interactions involved. The interpretation of \gls{libs} data is particularly complicated by factors such as multicollinearity among spectral features and matrix effects, which can obscure the relationships between the observed spectra and the actual chemical compositions.

Reviewer: We've mentioned multicollinearity and matrix effects in the introduction. This sentence reads a little like its the first time we've presented it. Does it perhaps make sense to rephrase it so that its clear that its not the first time we mention it?

The process of elemental quantification using \gls{libs} data is constrained by several factors:
\begin{itemize}
    \item \textbf{Data Dimensionality and Collinearity:} High dimensionality of spectral data coupled with multicollinearity, where multiple spectral features may exhibit strong correlations, complicates the modeling and analysis\cite{andersonImprovedAccuracyQuantitative2017}.

>     \item \textbf{Matrix Effects:} Used as a catch-all term that encompasses any effect that can cause the intensity of emission lines from an element to vary independent of that element's concentration. Different background materials can alter the emission intensities and pose significant challenges in accurately quantifying elemental concentrations. The spectra are complex due to the interaction of multiple physical processes including the coupling process between the laser photons and the target, self-absorption of optical emission lines within the plasma, recombination of elements into molecules, and collisional interactions in the plasma\cite{cleggRecalibrationMarsScience2017, andersonImprovedAccuracyQuantitative2017}.

Reviewer: ```
Used as a catch-all term that encompasses any effect that can cause the intensity of emission lines from an element to vary independent of that element's concentration.
```
This sentence feels very familiar. Are we 100% sure this is not a direct copy paste error?

Author: Yes, it's paraphrased from `andersonImprovedAccuracyQuantitative2017`:
![image](https://github.com/chhoumann/thesis-chemcam/assets/29108628/7e24a1c6-d94d-41ac-acfc-d28b4c9a8e76)

I cited them, but do you think it's too close?

Reviewer: Yes I think we need to rephrase ours. 
```suggestion
    \item \textbf{Matrix Effects:} This term refers to any effect that can cause variations in the intensity of emission lines of an element independent of the concentration of that element. Different background materials can alter the emission intensities and pose significant challenges in accurately quantifying elemental concentrations. The spectra are complex due to the interaction of multiple physical processes including the coupling process between the laser photons and the target, self-absorption of optical emission lines within the plasma, recombination of elements into molecules, and collisional interactions in the plasma\cite{cleggRecalibrationMarsScience2017, andersonImprovedAccuracyQuantitative2017}.
```

This is one possible rephrasing

Author: Sure!

    \item \textbf{Data Availability:} Due to the high cost of data collection, datasets are often small, which may limit the generalizability of the models\cite{p9_paper}.
\end{itemize}

These constraints necessitate the development of advanced computational models that can effectively account for and mitigate these complexities to improve the accuracy and reliability of chemical composition analysis using \gls{libs} data.

> The input to our computational models consists of \gls{libs} spectral data, which includes intensity readings across a spectrum of wavelengths. This data is in the form of Clean, Calibrated Spectra\cite{andersonImprovedAccuracyQuantitative2017}, the output of level 1 processing as described by \citet{wiensPreflightCalibrationInitial2013}. The wavelength intensities are quantified in units of \\photon/shot/mm\textsuperscript{2}/sr/nm.

Reviewer: ```suggestion
The input to our computational models consists of \gls{libs} spectral data, which includes intensity readings across a spectrum of wavelengths. This data is in the form of Clean, Calibrated Spectra\cite{andersonImprovedAccuracyQuantitative2017}, the output of level 1 processing as described by \citet{wiensPreflightCalibrationInitial2013}. The wavelength intensities are quantified in units of photon/shot/mm\textsuperscript{2}/sr/nm.
```

Seems to be a newline right before the units are introduced. I would remove it

Author: Line formatting was broken before, but now broken due to it. Lol. Should just postpone trying to fix that kind of stuff until we're done with the contents.

We have:

\begin{itemize}

>     \item \textbf{Matrix $A_{t \times o}$}: This matrix denotes the chemical concentrations in weight percent for oxides $o$ across targets $t$.

Reviewer: ```suggestion
    \item \textbf{Matrix $A_{to}$}: This matrix denotes the chemical concentrations in weight percent for oxides $o$ across targets $t$.
```
I dont think the current subscript is correct. Source:
https://en.wikipedia.org/wiki/Matrix_(mathematics)

Author: Your source seems to agree?

![image](https://github.com/chhoumann/thesis-chemcam/assets/29108628/97dc6d83-8265-4009-bfae-d91bfbe666d5)

Author: Updated 
![image](https://github.com/chhoumann/thesis-chemcam/assets/29108628/52ea521f-2da5-42e4-852c-0a97d3f494b6)

    \item \textbf{Matrix $B_{w \times s}$}: A Boolean matrix that links wavelengths $w$ to spectrometers $s$, indicating whether a specific wavelength is detected by a spectrometer.
    \item \textbf{Matrix $C_{t \times l \times s \times w}$}: Holds the spectral intensity data, where each entry represents the intensity recorded for a target $t$ at location $l$, for shot $s$, at wavelength $w$.
    \item \textbf{Matrix $D_{t \times l \times w}$}: Derived from matrix $C$ by averaging the intensities across shots to provide a clearer signal for each location and wavelength:
    \[
    D[t, l, w] = \frac{1}{|S|} \sum_{s \in S} C[t, l, s, w].
    \]
    \item \textbf{Matrix $E_{t \times l \times w}$}: The result of $D$ processed by applying wavelength-specific masks, setting intensities to zero in masking ranges to focus on relevant spectral features.
\end{itemize}

The outputs of our models are the quantified chemical compositions of geological samples. These are primarily the concentrations of major elements, represented in weight percentage. While trace elements are also present, our current analysis does not quantify these.

> Our goal is to construct a mapping function $\mathcal{F}: \mathbb{R}^N \rightarrow \mathbb{R}^O$, where $N$ represents the dimensionality of the processed \gls{libs} signals and $O$ represents the number of target oxides. This function maps a processed \gls{libs} signal vector $\mathbf{x} \in \mathbb{R}^N$ to a vector $\mathbf{v} \in \mathbb{R}^O$ of estimated oxide concentrations:

Reviewer: ```suggestion
Our goal is to construct a mapping function $\mathcal{F}: \mathbb{R}^N \rightarrow \mathbb{R}^O$, where $N$ represents the dimensionality of the processed \gls{libs} signals and $O$ represents the number of target oxides. This function maps a processed \gls{libs} signal vector $\mathbf{x} \in \mathbb{R}^N$ to a vector $\mathbf{v} \in \mathbb{R}^O$ of estimated oxide concentrations:
```
As discussed I feel like there needs to be a bridge between our matrices and this mapping function

\[
\mathbf{v} = \mathcal{F}(\mathbf{x}).
\]

> The primary metric used to evaluate the performance of our computational models is the \gls{rmse}. \gls{rmse} is calculated using the formula:

Reviewer: I think we should also include standard deviation. A think its a valid metric to use.

\[
\text{RMSE} = \sqrt{\frac{1}{n} \sum_{i=1}^{n} (\mathbf{v}_i - \hat{\mathbf{v}}_i)^2}
\]
where \( \mathbf{v}_i \) is the vector of actual oxide concentrations for the \( i \)-th sample, \( \hat{\mathbf{v}}_i \) is the corresponding vector of predicted oxide concentrations, and \( n \) is the total number of samples. This measure quantifies the average magnitude of the prediction error across all predicted values, providing a clear indication of model accuracy in quantifying chemical compositions.

> Our evaluation involves a combination of methodologies. Initially, the model is validated using a train/test split, where it is trained on a subset of data and tested on a separate subset to assess its predictive performance. Further, we compare the performance of our models against a baseline model previously established in our research \cite{p9_paper}, as well as against chemical analysis methods and other computational models documented in existing literature. This comprehensive comparison helps to underscore the improvements and benefits our approach offers over traditional methods and prior computational approaches.

Reviewer: I dont think this part fits here. I would expect the problem definition to be about what the problem is and why its important in general terms. This feels more like coming up with a solution. Well written though.

Author: Removed!

NASA's exploration of Mars, beginning with the Viking missions in the 1970s, has progressively deepened our understanding of Mars\cite{marsnasagov_vikings}. The \gls{msl} mission, which landed the Curiosity rover in Gale Crater in 2012, represents a pivotal step in this journey. Curiosity is equipped with the \gls{chemcam} instrument, a tool that uses \gls{libs} to analyze the chemical composition of Martian rocks and soils directly and non-invasively\cite{chemcamNasaWebsite}.

\gls{libs} is particularly suitable for the Martian environment because of its ability to perform rapid chemical analyzes remotely, creating a plasma that can be spectrally analyzed to determine the elemental composition of the vaporized material. This capability is crucial because it allows scientists to quickly and efficiently assess the geochemistry of multiple sites without physically moving the rover, thus conservatively managing the rover's limited energy and resources. The mission's focus has been on assessing past habitability, and the data gathered by \gls{chemcam} has been instrumental in identifying environments that could have supported life\cite{chemcamNasaWebsite,curiosityNasaWebsite}.

> The task of quantifying the oxides in Martian rock and soil samples begins with the \gls{libs} spectral data collected by Curiosity. This data comprises high-dimensional spectra with thousands of potential features, each corresponding to a specific element's emission lines. The computational challenge lies in accurately interpreting these complex data sets to deduce the concentrations of various elements, especially major oxides like iron, magnesium, and silicon, which are crucial for understanding Martian geology.

Reviewer: "... This data comprises high-dimensional spectra with thousands of potential features, each corresponding to a specific element's emission lines."

are they potential features? Would it not be more correct to just call them features?

Author: Potential features because you preprocess, modify, and otherwise mask/remove various aspects of the data & even entire dimensions. So using the more relaxed 'potential' is appropriate IMO.

Reviewer: Alright. I buy this argument. Resolved.

\section{Problem Definition}\label{sec:problem_definition}
The primary objective of this research is to enhance computational methods for the accurate and robust quantification of chemical compositions using \gls{libs} data.
This enhancement seeks to improve the adaptability and precision of \gls{libs} across various environmental contexts, including terrestrial and planetary geological samples.

Quantifying chemical compositions from \gls{libs} spectral data poses significant challenges due to the high dimensionality of the data and the complex physical and chemical interactions involved. The interpretation of \gls{libs} data is particularly complicated by factors such as multicollinearity among spectral features and matrix effects, which can obscure the relationships between the observed spectra and the actual chemical compositions.

> The process of elemental quantification using \gls{libs} data is constrained by several factors:

Reviewer 2: Unsure if you will address this with the comment above, but you repeat what you just said in the sentence prior to this:

> The interpretation of \gls{libs} data is particularly complicated by factors such as multicollinearity among spectral features and matrix effects, which can obscure the relationships between the observed spectra and the actual chemical compositions.

Maybe try and write these two sentences together instead and have it flow into the itemize?

\begin{itemize}
    \item \textbf{Data Dimensionality and Collinearity:} High dimensionality of spectral data coupled with multicollinearity, where multiple spectral features may exhibit strong correlations, complicates the modeling and analysis\cite{andersonImprovedAccuracyQuantitative2017}.

>     \item \textbf{Matrix Effects:} This term refers to any effect that can cause variations in the intensity of emission lines of an element independent of the concentration of that element. Different background materials can alter the emission intensities and pose significant challenges in accurately quantifying elemental concentrations. The spectra are complex due to the interaction of multiple physical processes including the coupling process between the laser photons and the target, self-absorption of optical emission lines within the plasma, recombination of elements into molecules, and collisional interactions in the plasma\cite{cleggRecalibrationMarsScience2017, andersonImprovedAccuracyQuantitative2017}.

Reviewer 2: __
```suggestion
    \item \textbf{Matrix Effects:} This term refers to any effect that can cause variations in the intensity of emission lines of an element independent of the concentration of that element. The presence of different non-major oxides can alter the emission intensities and pose significant challenges in accurately quantifying elemental concentrations. The spectra are complex due to the interaction of multiple physical processes including the coupling process between the laser photons and the target, self-absorption of optical emission lines within the plasma, recombination of elements into molecules, and collisional interactions in the plasma\cite{cleggRecalibrationMarsScience2017, andersonImprovedAccuracyQuantitative2017}.
```
If I understand correctly, you're explaining how the presence of other elements can affect the spectra, in which case this is a bit more specific.

>     \item \textbf{Data Availability:} Due to the high cost of data collection, datasets are often small, which may limit the generalizability of the models\cite{p9_paper}.

Reviewer 2: __
```suggestion
    \item \textbf{Data Availability:} Due to the high cost of data collection, datasets are often small, which can limit the generalizability and robustness of the models\cite{p9_paper}.
```
Wouldn't this be a good place to bring up robustness and why it's difficult to achieve robust models?

\end{itemize}

> These constraints necessitate the development of advanced computational models that can effectively account for and mitigate these complexities to improve the accuracy and reliability of chemical composition analysis using \gls{libs} data.

Reviewer 2: ```suggestion
These challenges necessitate the development of advanced computational models that can effectively account for and mitigate the complexities to improve the accuracy and reliability of chemical composition analysis using \gls{libs} data.
```

> The input to our computational models consists of \gls{libs} spectral data, which includes intensity readings across a spectrum of wavelengths. This data is in the form of Clean, Calibrated Spectra\cite{andersonImprovedAccuracyQuantitative2017}, the output of level 1 processing as described by \citet{wiensPreflightCalibrationInitial2013}. The wavelength intensities are quantified in units of photon/shot/mm\textsuperscript{2}/sr/nm.

Reviewer 2: I don't like how we jump from being general: "these constraints necessitate the development of advanced computational models" 
And now we are suddenly concrete: "The input to our models". Either this paragraph needs to follow the same general thread as previous (which it probably should, considering this is the problem definition and it is likely too early to start talking about our developed models), or we need to bridge the gap between these to paragraphs in a cleaner way.

As an example of a generalized phrasing, we could write something like this instead:
>The input to such computational models consists of \gls{libs} spectral data, which includes intensity readings across a spectrum of wavelengths. This data is in the form of Clean, Calibrated Spectra\cite{andersonImprovedAccuracyQuantitative2017}, the output of level 1 processing as described by \citet{wiensPreflightCalibrationInitial2013}. The wavelength intensities are quantified in units of photon/shot/mm\textsuperscript{2}/sr/nm.

I just changed one word (our -> such) to make it generalized, but I think something like that would work much better considering the aforementioned points.

Author: Agree. Changed.

> We have:

Reviewer 2: If this is meant to serve as a contintuation of the previous paragraph, perhaps we should change this to "Formally:", or "Formally, we have:" to make it more clear. Right now it doesn't flow too well with the previous paragraph.

Author: Changed

\begin{itemize}

>     \item \textbf{Matrix $A[t, o]$}: This matrix denotes the chemical concentrations in weight percent for oxides $o$ across targets $t$.

Reviewer 2: ```suggestion
    \item \textbf{Matrix $A[t, o]$}: This matrix denotes the chemical concentrations in weight percentage for oxides $o$ across targets $t$.
```

Author: Clegg et al. uses `weight percent`

    \item \textbf{Matrix $B[w, s]$}: A Boolean matrix that links wavelengths $w$ to spectrometers $s$, indicating whether a specific wavelength is detected by a spectrometer.
    \item \textbf{Matrix $C[t, l, s, w]$}: Holds the spectral intensity data, where each entry represents the intensity recorded for a target $t$ at location $l$, for shot $s$, at wavelength $w$.
    \item \textbf{Matrix $D[t, l, w]$}: Derived from matrix $C$ by averaging the intensities across shots to provide a clearer signal for each location and wavelength:
    \[
    D[t, l, w] = \frac{1}{|S|} \sum_{s \in S} C[t, l, s, w].
    \]
    \item \textbf{Matrix $E[t, l, w]$}: The result of $D$ processed by applying wavelength-specific masks, setting intensities to zero in masking ranges to focus on relevant spectral features.

> \end{itemize}

Reviewer 2: If this is the first time we are using $o$, $t$, $w$ and $s$, we need to define them. Are they sets? Matrices? This is not clear to the reader as it stands right now.

Author: Done!

> The outputs of our models are the quantified chemical compositions of geological samples. These are primarily the concentrations of major elements, represented in weight percentage. While trace elements are also present, our current analysis does not quantify these.

Reviewer 2: We call it "trace elements" here - have we introduced this concept to the reader before this part? Does the reader know that there are other elements present too? See my previous comment about rephrasing the itemize point about Matrix Effects.

\[
\mathbf{v} = \mathcal{F}(\mathbf{x}).
\]

The primary metric used to evaluate the performance of our computational models is the \gls{rmse}. \gls{rmse} is calculated using the formula:
\[
\text{RMSE} = \sqrt{\frac{1}{n} \sum_{i=1}^{n} (\mathbf{v}_i - \hat{\mathbf{v}}_i)^2}
\]
where \( \mathbf{v}_i \) is the vector of actual oxide concentrations for the \( i \)-th sample, \( \hat{\mathbf{v}}_i \) is the corresponding vector of predicted oxide concentrations, and \( n \) is the total number of samples. This measure quantifies the average magnitude of the prediction error across all predicted values, providing a clear indication of model accuracy in quantifying chemical compositions.

Additionally, we will evaluate the robustness of our models using the standard deviation of prediction errors, defined as:
\[
\sigma_{error} = \sqrt{\frac{1}{n-1} \sum_{i=1}^{n} (e_i - \bar{e})^2}
\]
where \( e_i = \mathbf{v}_i - \hat{\mathbf{v}}_i \) and \( \bar{e} \) is the mean error. This metric assesses the variability of the prediction errors, offering insights into the consistency and reliability of the model's performance. A lower standard deviation indicates a more robust model.

> Success will be evaluated primarily through these metrics, comparing the predictive accuracy and robustness of our models against existing benchmarks and baseline models established in prior research.

Reviewer 2: Should we mention anything about cross-validation here? Or is that too implementation specific to include in the problem definition?

Author: Too specific IMO. Belongs in methodology.

> \textbf{Problem Definition:} This thesis aims to address the challenges in predicting major oxide compositions from \gls{libs} data by enhancing computational methods to improve accuracy and robustness. We propose to develop computational models capable of effectively accounting for and mitigating the complexities inherent in \gls{libs} data. Our models will take as input a matrix in the form of $E$, as well as ground truth data in the form of $A$, to construct a mapping function $\mathcal{F}: \mathbb{R}^N \rightarrow \mathbb{R}^O$, mapping processed \gls{libs} signals to estimated oxide concentrations. Success will be evaluated primarily through the \gls{rmse} metric, comparing the predictive accuracy of our models against existing benchmarks and baseline models established in prior research.

Reviewer 2: > Success will be evaluated primarily through the \gls{rmse} metric, comparing the predictive accuracy of our models against existing benchmarks and baseline models established in prior research.

is the exact same phrasing as two lines above. Should we phrase it more concisely here instead of copying the sentence?

The task of quantifying the oxides in Martian rock and soil samples begins with the \gls{libs} spectral data collected by Curiosity. This data comprises high-dimensional spectra with thousands of potential features, each corresponding to a specific element's emission lines. The computational challenge lies in accurately interpreting these complex data sets to deduce the concentrations of various elements, especially major oxides like iron, magnesium, and silicon, which are crucial for understanding Martian geology.

> Computational models take these spectral readings as input and process them through several stages of analysis. Initially, the data undergoes preprocessing to correct for any instrumental effects and to calibrate the raw spectra. This step ensures that the readings are accurate and can be reliably used for quantitative analysis. 

Reviewer 2: First sentence feels like a repetition of what we have already said, maybe start with "As mentioned" or rephrase it.

Author: Added

> The cleaned data is fed into machine learning models, which have been trained to correlate specific spectral signatures with known concentrations of elements, based on a database of Earth-based and synthetic Martian analogs.

Reviewer 2: "fed into" is quite an informal way of saying "used as input"

Author: Changed

> The output of this process is a detailed quantitative analysis of the soil or rock sample's chemical composition, expressed in weight percentages of each detected oxide\cite{wiensPreflightCalibrationInitial2013, cleggRecalibrationMarsScience2017}.

Reviewer 2: I don't know if it's just me, but these 3 lines feel really similar to what we've already said earlier. They repeat a lot of concepts we've already introduced. I feel like this could either be cut entirely or rephrased to much more concisely reiteate the main points.

\citet{cleggRecalibrationMarsScience2017} undertook this task and created a pipeline for predicting the concentration of oxides in Martian soil samples, referred to as \gls{moc}.

> More recently, in 2022, the Perseverance rover landed on Mars, equipped with advanced instruments designed to continue the exploration and analysis of the Martian surface. This rover also uses a \gls{libs} instrument, called SuperCam. This instrument is the successor to \gls{chemcam} and shows the continued success of the \gls{libs} technique in the Martian environment. The Perseverance mission led to continued research in the realm of quantification based on \gls{libs} data\cite{andersonPostlandingMajorElement2022}, showing it continues to be an important area of research.

Reviewer 2: ```suggestion
More recently, in 2022, the Perseverance rover landed on Mars, equipped with advanced instruments designed to continue the exploration and analysis of the Martian surface. This rover also uses a \gls{libs} instrument, called SuperCam. This instrument is the successor to \gls{chemcam} and shows the continued success of the \gls{libs} technique in the Martian environment. The Perseverance mission has spurred ongoing research in elemental quantification using \gls{libs} data\cite{andersonPostlandingMajorElement2022}, demonstrating its continued importance in the field.
```

Author: ```suggestion
More recently, in 2022, the Perseverance rover landed on Mars, equipped with advanced instruments designed to continue the exploration and analysis of the Martian surface. This rover also uses a \gls{libs} instrument, called SuperCam. This instrument is the successor to \gls{chemcam} and shows the continued success of the \gls{libs} technique in the Martian environment. The Perseverance mission highlighted the ongoing research effort in developing elemental quantification models using \gls{libs} data\cite{andersonPostlandingMajorElement2022}, demonstrating its continued importance as a research field.
```
